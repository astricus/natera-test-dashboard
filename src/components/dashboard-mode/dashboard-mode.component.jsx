import React from 'react';
import { connect } from 'react-redux';
import { selectMode } from '../../redux/mode/mode.selectors';
import CompanyInfo from '../company-info/company-info.component';
import Highlights from '../highlights/highlights.component';
import Notification from '../notifications/notification.component';
import ProjectsStatuses from '../projects-statuses/projects-statuses.component';
import Footer from '../footer/footer.component';

import './dashboard-mode.styles.scss';

const DashboardMode = ({ mode }) => (
  <div className={`dashboard-mode ${!mode ? 'd-non' : ''}`}>
    <CompanyInfo />
    {/* Mode prop needed to be passed directly as true because animation take some time and <Highlights> is a reusable component. 
      Passing Redux state.mode.mode as a prop or using mapStateToProps inside this component and it's children will case this component update 
      when mode animation is triggered. */}
    <Highlights mode={true} />
    <Notification />
    {/* Mode prop needed to be passed directly as true because animation take some time and <ProjectsStatuses> is a reusable component. 
      Passing Redux state.mode.mode as a prop or using mapStateToProps inside this component and it's children will case this component update 
      when mode animation is triggered. */}
    <ProjectsStatuses key={`projects-statuses__${mode}`} mode={true} />
    <Footer />
  </div>
);

const mapStateToProps = (state) => ({
  mode: selectMode(state),
});

export default connect(mapStateToProps)(DashboardMode);
