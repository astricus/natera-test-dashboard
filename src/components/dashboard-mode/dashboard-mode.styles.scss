// Media-queries mixins
@import '../../styles/mixins.scss';

.dashboard-mode {
  width: 100%;
  height: auto;
  // transition: transform 0.2s ease-in;
  // transform: translateX(0);

  @include respond(phone) {
    position: relative;
  }
}

.translate-to-edit-mode {
  transform: translateX(-102%);
}
