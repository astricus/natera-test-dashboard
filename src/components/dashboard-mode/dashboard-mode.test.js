import React from 'react';
import { shallow } from 'enzyme';
import { DashboardMode } from './dashboard-mode.component';

describe('DashboardMode component', () => {
  const dashboardModeIsVisible = shallow(<DashboardMode mode={true} />);
  const dashboardModeNotVisible = shallow(<DashboardMode mode={false} />);

  test('should render DashboardMode component', () => {
    expect(dashboardModeIsVisible).toMatchSnapshot();
  });

  test('should render DashboardMode component', () => {
    expect(dashboardModeNotVisible).toMatchSnapshot();
  });
});
