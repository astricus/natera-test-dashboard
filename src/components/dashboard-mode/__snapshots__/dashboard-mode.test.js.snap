// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardMode component should render DashboardMode component 1`] = `
<div
  className="dashboard-mode "
>
  <CompanyInfo />
  <Connect(Highlights)
    mode={true}
  />
  <Connect(Notification) />
  <ProjectsStatuses
    key="projects-statuses__true"
    mode={true}
  />
  <Footer />
</div>
`;

exports[`DashboardMode component should render DashboardMode component 2`] = `
<div
  className="dashboard-mode d-non"
>
  <CompanyInfo />
  <Connect(Highlights)
    mode={true}
  />
  <Connect(Notification) />
  <ProjectsStatuses
    key="projects-statuses__false"
    mode={true}
  />
  <Footer />
</div>
`;
